plan/refiner.go

```diff
@@ -437,7 +436,43 @@ func (c *conditionChecker) checkScalarFunction(scalar *expression.ScalarFunction
 	case ast.IsNull, ast.IsTruth, ast.IsFalsity:
 		return c.checkColumn(scalar.Args[0])
 	case ast.UnaryNot:
+		// Don't support "not like" and "not in" convert to access conditions.
+		if s, ok := scalar.Args[0].(*expression.ScalarFunction); ok {
+			if s.FuncName.L == ast.In || s.FuncName.L == ast.Like {
+				return false
+			}
+		}
 		return c.newCheck(scalar.Args[0])
+	case ast.In:
+		if !c.checkColumn(scalar.Args[0]) {
+			return false
+		}
+		for _, v := range scalar.Args[1:] {
+			if _, ok := v.(*expression.Constant); !ok {
+				return false
+			}
+		}
+		return true
+	case ast.Like:
+		if !c.checkColumn(scalar.Args[0]) {
+			return false
+		}
+		pattern, ok := scalar.Args[1].(*expression.Constant)
+		if !ok {
+			return false
+		}
+		if pattern.Value.IsNull() {
+			return false
+		}
+		patternStr, err := pattern.Value.ToString()
+		if err != nil {
+			return false
+		}
+		if len(patternStr) == 0 {
+			return true
+		}
+		firstChar := patternStr[0]
+		return firstChar != '%' && firstChar != '.'
```

Comment:
select a from t where c like 'abc%de'
The like expr could be converted to accesss condition. But it should also be added into filter condition.